components/base/kinematicbase/kinematics.go

```diff
@@ -25,6 +25,8 @@ type KinematicBase interface {
 	// ErrorState takes a complete motionplan, as well as the index of the currently-executing set of inputs, and computes the pose
 	// difference between where the robot in fact is, and where it ought to be.
 	ErrorState(context.Context, [][]referenceframe.Input, int) (spatialmath.Pose, error)
+
+	FrameSystem() referenceframe.FrameSystem
```

Comment:
I don't think this addition is necessary. We can just do the replacement at the motion service level, and then there won't be changes needed for any of the files in `kinematicbase`